webapp/src/ts/providers/parse.provider.ts

```diff
@@ -156,11 +157,20 @@ class ASTCompiler {
     return 'ctx';
   }
 
-  processLiteralPrimitive() {
-    const ast = this.cAst;
+  processLiteralPrimitive(ast) {
     return isString(ast.value) ? `"${ast.value}"` : ast.value;
   }
 
+  processUnaryLiteralPrimitive(ast) {
+    const literalValue = this.processLiteralPrimitive(ast.expr);
+
+    if (ast.operator === '-') {
+      return isString(literalValue) ? '-' + literalValue : (literalValue * -1);
```

Comment:
Do we need a unit test to cover the `isString... === true` side of this ternary?  The only time I see this line hit in the current parse tests, `literalValue` is a number...

Reply:
That scenario results in `NaN`, I've added unit tests to cover it.